File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        values = []
        operations = {
            '+': lambda a, b: a + b,
            '-': lambda a, b: a - b,
            '*': lambda a, b: a * b,
            '/': lambda a, b: int(a / b)
        }

        for token in tokens:
            if token in operations:
                perform = operations[token]
                b, a = values.pop(), values.pop()
                values.append(perform(a, b))
            else:
                values.append(int(token))
        
        return values.pop()

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.js
const evalRPN = (tokens) => {
    const values = [];
    const operations = {
    '+': (a, b) => a + b,
    '-': (a, b) => a - b,
    '*': (a, b) => a * b,
    '/': (a, b) => Math.trunc(a / b)
    };

    for (const token of tokens) {
        if (token in operations) {
            const [b, a] = [values.pop(), values.pop()];
            const process = operations[token];
            values.push(process(a, b));
        } else {
            values.push(parseInt(token));
        }
    }

    return values.pop();
};
